lte/gateway/c/core/oai/test/spgw_task/test_spgw_procedures_with_injected_state.cpp

```diff
@@ -0,0 +1,149 @@
+#include <gtest/gtest.h>
+#include <string>
+#include <thread>
+
+#include "../mock_tasks/mock_tasks.h"
+#include "spgw_test_util.h"
+#include "spgw_state.h"
+#include "spgw_state_converter.h"
+#include "mock_spgw_op.h"
+
+extern "C" {
+#include "mme_config.h"
+#include "pgw_handlers.h"
+#include "sgw_context_manager.h"
+#include "sgw_defs.h"
+#include "sgw_handlers.h"
+#include "spgw_config.h"
+}
+
+extern bool hss_associated;
+
+namespace magma {
+namespace lte {
+
+task_zmq_ctx_t task_zmq_ctx_main_spgw;
+
+static int handle_message(zloop_t* loop, zsock_t* reader, void* arg) {
+  MessageDef* received_message_p = receive_msg(reader);
+
+  switch (ITTI_MSG_ID(received_message_p)) {
+    default: { } break; }
+
+  itti_free_msg_content(received_message_p);
+  free(received_message_p);
+  return 0;
+}
+
+class SPGWAppInjectedStateProcedureTest : public ::testing::Test {
+  virtual void SetUp() {
+    // setup mock MME app task
+    mme_app_handler = std::make_shared<MockMmeAppHandler>();
+    itti_init(
+        TASK_MAX, THREAD_MAX, MESSAGES_ID_MAX, tasks_info, messages_info, NULL,
+        NULL);
+
+    // initialize configs
+    mme_config_init(&mme_config);
+    spgw_config_init(&spgw_config);
+    create_partial_lists(&mme_config);
+    mme_config.use_stateless = false;
+    hss_associated           = true;
+
+    task_id_t task_id_list[2] = {TASK_MME_APP, TASK_SPGW_APP};
+    init_task_context(
+        TASK_MAIN, task_id_list, 2, handle_message, &task_zmq_ctx_main_spgw);
+
+    std::thread task_mme_app(start_mock_mme_app_task, mme_app_handler);
+    task_mme_app.detach();
+
+    std::cout << "Running setup" << std::endl;
+    // initialize the SPGW task
+    // might need to mock this function
+    spgw_state_init(false, &spgw_config);    
```

Comment:
As mentioned above, let's call `spgw_app_init` instead of mocking it, and call `mock_read_spgw_ue_state_db` separately after initializing SPGW app.